api/http/handler/backup/handler.go

```diff
@@ -35,8 +36,8 @@ func NewHandler(bouncer *security.RequestBouncer, dataStore portainer.DataStore,
 		adminMonitor:    adminMonitor,
 	}
 
-	h.Handle("/backup", bouncer.RestrictedAccess(adminAccess(httperror.LoggerHandler(h.backup)))).Methods(http.MethodPost)
-	h.Handle("/restore", bouncer.PublicAccess(httperror.LoggerHandler(h.restore))).Methods(http.MethodPost)
+	h.Handle("/backup", restrictDemoEnv(isDemo, bouncer.RestrictedAccess(adminAccess(httperror.LoggerHandler(h.backup))))).Methods(http.MethodPost)
+	h.Handle("/restore", restrictDemoEnv(isDemo, bouncer.PublicAccess(httperror.LoggerHandler(h.restore)))).Methods(http.MethodPost)
```

Comment:
If changes in this pr aren't going to escape the `demo` branch, then we can simply replace the middleware chain with the simplified version of the `restrictDemoEnv` handler. Such handler can always return an error.

Reply:
I believe we do want to be on the main branch soon (behind `--demo` flag), but @yi-portainer  wanted this for short term 

It's better to leave the codebase as similar as possible so it will make it easy to sync them